### PR TITLE
Reuse PerpetualContracts for Deploy and Benchmark scripts

### LINKED ISSUE #6: Remove perpetual contracts and inherit from the deploy script
- We redefine the same contract in the deploy scripts and the tests and we should define them in a single place.

_Originally posted by @mds1 in https://github.com/ScopeLift/optimizoors-perpetual-protocol/pull/1#discussion_r1177015948_

### PR DESCRIPTION
**Pull request by alexkeating**
#### Description

- Reuse PerpetualContracts for Benchmark and Deploy scripts

Closes #6

### CODE REVIEW

File: script/Deploy.s.sol [hunk 1]
```diff
@@ -1,21 +1,14 @@
 // SPDX-License-Identifier: UNLICENSED
-
-pragma solidity ^0.8.16;
+pragma solidity >=0.8.0;
```
Comment (mds1) [inline script/Deploy.s.sol:2]: Same comment around pragmas as in #13, in general you don't want `>=` unless it's an interface file that supports a range of versions. We don't know whether or not this contract will be compatible with solidity `0.9.0` so we don't want to imply it does with the pragma

Comment (alexkeating) [reply script/Deploy.s.sol:2]: Ah okay makes sense, I updated all the pragmas in 13.

Comment (mds1) [reply script/Deploy.s.sol:2]: Ok, I'm assuming we'll merge the PRs in order so I'll leave this unresolved until 13 is merged + this rebased on top, to make sure it didn't get reverted in the rebase

Comment (alexkeating) [reply script/Deploy.s.sol:2]: Sounds good to me

Comment (alexkeating) [reply script/Deploy.s.sol]: Rebased with #13 in main

**Review by mds1 — APPROVED** [review]

**Pull request merged**